modules/tree/main/index.coffee

```diff
@@ -10,33 +40,44 @@ formatIcon = (node, iconElement, animate) =>
         .go(true)
 
 format = (tree, element, animate) ->
+  treeNode = hx.select(element)
+  treeNodeComponent = treeNode.component()
+
+  renderLazyChildren = (rootNode, recursive) ->
+    if rootNode.component()? and rootNode.selectAll('.hx-tree-node').empty()
+      rootNode.component().renderChildren()
+      formatChildren(tree, rootNode.selectAll('.hx-tree-node'), animate)
```

Comment:
Detect if a node has children to render (i.e. it has `renderChildren` but doesn't have any `hx-tree-node`s) and renders them.
`formatChildren` is a helper so we don't have to refresh the whole tree, only the bit that's changed